hedera-node/hedera-app/src/main/java/com/hedera/node/app/bbm/tokentypes/TokenTypesDumpUtils.java

```diff
@@ -0,0 +1,322 @@
+            Pair.of("accountsKycGrantedByDefault", getFieldFormatter(Token::accountsKycGrantedByDefault, booleanFormatter)),
+            Pair.of("treasuryAccount", getFieldFormatter(Token::treasury, getNullableFormatter(ThingsToStrings::toStringOfEntityId))),
+            Pair.of("autoRenewAccount", getFieldFormatter(Token::autoRenewAccount, getNullableFormatter(ThingsToStrings::toStringOfEntityId))),
+            Pair.of("feeSchedule", getFieldFormatter(Token::feeSchedule,
+                    getNullableFormatter(getListFormatter(ThingsToStrings::toStringOfFcCustomFee, SUBFIELD_SEPARATOR)))),
+            Pair.of("adminKey", getFieldFormatter(Token::adminKey, getOptionalFormatter(ThingsToStrings::toStringOfJKey))),
+            Pair.of("feeScheduleKey", getFieldFormatter(Token::feeScheduleKey, getOptionalFormatter(ThingsToStrings::toStringOfJKey))),
+            Pair.of("frezeKey", getFieldFormatter(Token::freezeKey, getOptionalFormatter(ThingsToStrings::toStringOfJKey))),
+            Pair.of("kycKey", getFieldFormatter(Token::kycKey, getOptionalFormatter(ThingsToStrings::toStringOfJKey))),
+            Pair.of("pauseKey", getFieldFormatter(Token::pauseKey, getOptionalFormatter(ThingsToStrings::toStringOfJKey))),
+            Pair.of("supplyKey", getFieldFormatter(Token::supplyKey, getOptionalFormatter(ThingsToStrings::toStringOfJKey))),
+            Pair.of("wipeKey", getFieldFormatter(Token::wipeKey, getOptionalFormatter(ThingsToStrings::toStringOfJKey))));
+    // spotless:on
+
+    public static void dumpModTokenType(
+            @NonNull final Path path,
+            @NonNull final VirtualMap<OnDiskKey<TokenID>, OnDiskValue<com.hedera.hapi.node.state.token.Token>> tokens,
+            @NonNull final DumpCheckpoint checkpoint) {
+
+        try (@NonNull final var writer = new Writer(path)) {
+            final var allTokens = gatherTokensFromMod(tokens, Token::fromMod);
+            dump(writer, allTokens);
+            System.out.printf(
+                    "=== mod tokens report is %d bytes at checkpoint %s%n", writer.getSize(), checkpoint.name());
+        }
+    }
+
+    public static void dumpMonoTokenType(
+            @NonNull final Path path,
+            @NonNull final MerkleMap<EntityNum, MerkleToken> tokens,
+            @NonNull final DumpCheckpoint checkpoint) {
+        try (@NonNull final var writer = new Writer(path)) {
+            final var allTokens = gatherTokensFromMono(tokens, Token::fromMono);
+            dump(writer, allTokens);
+            System.out.printf(
+                    "=== mono tokens report is %d bytes at checkpoint %s%n", writer.getSize(), checkpoint.name());
+        }
+    }
+
+    @NonNull
+    private static Map<TokenType, Map<Long, Token>> gatherTokensFromMono(
+            @NonNull final MerkleMap<EntityNum, MerkleToken> source,
+            @NonNull final Function<MerkleToken, Token> valueMapper) {
```

Comment:
There is no need to pass a Function parameter if we are not reusing the gather method. Same for `gatherTokensFromMod`.